src/game.cpp

```diff
@@ -877,8 +878,22 @@ void game::load_npcs()
         if (temp->marked_for_death) {
             temp->die( nullptr );
         } else {
-            if (temp->my_fac != NULL)
+            if( temp->my_fac != NULL ) {
                 temp->my_fac->known_by_u = true;
+            }
+            if( temp->get_id() == 0 ) {
+                temp->set_id( assign_creature_id() );
+            }
+
+            const Creature *other = critter_with_id( temp->get_id() );
+            while( other != nullptr && other != temp ) {
```

Comment:
Strange, I have seen the same code before, properly in `Creature_tracker::add`. Why are you repeating this here?